dubbo-kubernetes/src/main/java/org/apache/dubbo/registry/kubernetes/KubernetesMeshEnvListener.java

```diff
@@ -134,42 +139,48 @@ private void subscribeDr(String appName) {
 
         try {
             Watch watch = kubernetesClient
-                .customResource(
-                    MeshConstant.getDrDefinition())
-                .watch(namespace, appName, null, new ListOptionsBuilder().build(), new Watcher<String>() {
-                    @Override
-                    public void eventReceived(Action action, String resource) {
-                        logger.info("Received VS Rule notification. AppName: " + appName + " Action:" + action + " Resource:" + resource);
+                    .genericKubernetesResources(
+                            MeshConstant.getDrDefinition())
+                    .inNamespace(namespace)
+                    .withName(appName)
+                    .watch(new Watcher<GenericKubernetesResource>() {
+                        @Override
+                        public void eventReceived(Action action, GenericKubernetesResource resource) {
+                            if (logger.isInfoEnabled()) {
+                                logger.info("Received VS Rule notification. AppName: " + appName + " Action:" + action + " Resource:" + resource);
+                            }
 
-                        if (action == Action.ADDED || action == Action.MODIFIED) {
-                            Map drRuleMap = new Gson().fromJson(resource, Map.class);
-                            String drRule = new Yaml(new SafeConstructor()).dump(drRuleMap);
+                            if (action == Action.ADDED || action == Action.MODIFIED) {
+                                Map drRuleMap = new Gson().fromJson(resource.getMetadata().getName(), Map.class);
```

Comment:
'drRuleMap' seems to be a rule but `resource.getMetadata().getName()` seems to be only a name.